ethclient/ethclient.go

```diff
@@ -141,6 +143,16 @@ func (ec *client) Client() *rpc.Client {
 
 // Blockchain Access
 
+// ChainConfig retrieves the current chain config.
+func (ec *client) ChainConfig(ctx context.Context) (*params.ChainConfigWithUpgradesJSON, error) {
+	var result *params.ChainConfigWithUpgradesJSON
```

Comment:
Can we return `*params.ChainConfig` instead of `*params.ChainConfigWithUpgradesJSON` since it's meant to be an intermediate type for marshalling?

Reply:
Sure thing. Should we also return `*params.ChainConfig` interally? https://github.com/ava-labs/subnet-evm/blob/master/internal/ethapi/api.go#L145

Reply:
It's also used in `getChainConfig` API and it also returns `ChainConfigWithUpgradesJSON`. we should keep them same.

Reply:
Fixed now.